contracts/contracts/ccip/onramp/contract.tolk

```diff
@@ -44,9 +44,15 @@ fun onInternalMessage(in: InMessage) {
             assert (in.senderAddress == feeQuoterAddress, Error.Unauthorized);
             onMessageValidationFailed(msg)
         }
-        // Permissionless
+        // Sender must be the Router
         OnRamp_Send => {
-            onSend(msg, in.senderAddress)
+            var st = lazy OnRamp_Storage.load();
+            val ccipSend = msg.msg.load();
+            // validate allowlist
+            val destChainConfig = st.destChainConfigs.mustGet(ccipSend.destChainSelector, Error.UnknownDestChainSelector as int);
+            // ccipSend must be forwarded from the router
+            assert(destChainConfig.router == in.senderAddress, Error.Unauthorized);
+            onSend(st, destChainConfig, msg)
```

Comment:
imo this just breaks the func in two as now some checks are in front, some are kept inside a func, and func args are now arbitrary based on what makes sense based on the checks which are extracted upfront